src/saveload/marker.rs

```diff
@@ -357,57 +366,65 @@ pub trait MarkerAllocator<M: Marker>: Resource {
     fn maintain(&mut self, _entities: &EntitiesRes, _storage: &ReadStorage<M>);
 }
 
-/// Basic marker implementation usable for saving and loading
+/// Basic marker implementation usable for saving and loading, uses `u64` as identifier
 #[derive(Clone, Copy, Debug, Hash, PartialEq, Eq, PartialOrd, Ord, Serialize, Deserialize)]
-pub struct U64Marker(u64);
+pub struct SimpleMarker<T: ?Sized>(u64, PhantomData<T>);
 
-impl Component for U64Marker {
+impl<T: ?Sized + Send + Sync + 'static> Component for SimpleMarker<T> {
     type Storage = DenseVecStorage<Self>;
 }
 
-impl Marker for U64Marker {
-    type Allocator = U64MarkerAllocator;
+impl<T> Marker for SimpleMarker<T>
+where
+    T: 'static + ?Sized + Clone + Debug + Eq + Hash + Send + Sync,
```

Comment:
According to https://doc.rust-lang.org/std/marker/struct.PhantomData.html, those traits are implemented by PhantomData when `T: ?Sized`, except it wouldn't compile, mentioning that `T` needed to be `Clone + Debug + ...`

Is there a fix to this? I assume I'm doing something wrong

Reply:
Oh, should it be
`where SimpleMarker<T>: Clone + Debug + Eq + Hash + Send + Sync` instead?

Reply:
Oh, I think I'm slamming my head against this: https://github.com/rust-lang/rust/issues/26925